cardano-cli/src/Cardano/CLI/EraBased/Governance/Actions/Run.hs

```diff
@@ -302,90 +304,51 @@ runGovernanceActionCreateProtocolParametersUpdateCmd
-
-        UpdateProtocolParametersConwayOnwards
-          _cOnwards
-          network
-          deposit
-          returnAddr
-          proposalUrl
-          proposalHash
-          checkProposalHash
-          mPrevGovActId
-          mConstitutionalScriptHash <-
-          fromExceptTCli $
-            hoistMaybe (GovernanceActionsValueUpdateProtocolParametersNotFound anyEra) $
-              uppConwayOnwards eraBasedPParams'
-
-        eraBasedPParams <- fromExceptTCli theUpdate
-
-        depositStakeCredential <-
-          getStakeCredentialFromIdentifier returnAddr
-
-        let updateProtocolParams = createEraBasedProtocolParamUpdate sbe eraBasedPParams
-
-            prevGovActId = L.maybeToStrictMaybe $ L.GovPurposeId <$> mPrevGovActId
-            proposalAnchor =
-              L.Anchor
-                { L.anchorUrl = unProposalUrl proposalUrl
-                , L.anchorDataHash = proposalHash
-                }
-
-        fromExceptTCli $ carryHashChecks checkProposalHash proposalAnchor ProposalCheck
-
-        let govAct =
-              UpdatePParams
-                prevGovActId
-                updateProtocolParams
-                (toShelleyScriptHash <$> L.maybeToStrictMaybe mConstitutionalScriptHash)
-
-        let proposalProcedure = createProposalProcedure sbe network deposit depositStakeCredential govAct proposalAnchor
-
-        fromEitherIOCli $
-          conwayEraOnwardsConstraints conwayOnwards $
-            writeFileTextEnvelope oFp (Just "Update protocol parameters proposal") proposalProcedure
-    )
-    sbe
+  let era = uppShelleyBasedEra eraBasedPParams'
+      sbe = convert era
+      oFp = uppFilePath eraBasedPParams'
+      UpdateProtocolParametersConwayOnwards
+        _cOnwards
+        network
+        deposit
+        returnAddr
+        proposalUrl
+        proposalHash
+        checkProposalHash
+        mPrevGovActId
+        mConstitutionalScriptHash = uppConwayOnwards eraBasedPParams'
+
+  eraBasedPParams <- fromExceptTCli theUpdate
+
+  depositStakeCredential <-
+    getStakeCredentialFromIdentifier returnAddr
+
+  let updateProtocolParams = createEraBasedProtocolParamUpdate sbe eraBasedPParams
+      prevGovActId = L.maybeToStrictMaybe $ L.GovPurposeId <$> mPrevGovActId
+      proposalAnchor =
+        L.Anchor
+          { L.anchorUrl = unProposalUrl proposalUrl
+          , L.anchorDataHash = proposalHash
+          }
+
+  fromExceptTCli $ carryHashChecks checkProposalHash proposalAnchor ProposalCheck
+
+  let govAct =
+        UpdatePParams
+          prevGovActId
+          updateProtocolParams
+          (toShelleyScriptHash <$> L.maybeToStrictMaybe mConstitutionalScriptHash)
+
+  let proposalProcedure = createProposalProcedure sbe network deposit depositStakeCredential govAct proposalAnchor
+
+  fromEitherIOCli @(FileError ()) $
+    obtainCommonConstraints era $
+      writeFileTextEnvelope oFp (Just "Update protocol parameters proposal") proposalProcedure
```

Comment:
It seems this is really close to `conwayProtocolParametersUpdate` now. Would it be possible to use `conwayProtocolParametersUpdate` here instead? Maybe moving it to this module. Or is there a reason why that is a bad idea that I am missing?

Reply:
It's the same but the plan is to further abstract this to make it clear it's parameterized on `Era era` and reuse it in the compatiblility functions.